File: Ejercicio 1/python_poligonos/poligonos.py
import random
import math
import sys
import time

inicioEjecucion = time.time() #Inicio de Ejecucion

random.seed() #Variables para Generar los Angulos
radius = 200
puntos = int(sys.argv[2])
angulo = 360 / puntos
cont = 0
angulos = []

angulos.append(cont)
for i in range(int(puntos)):
    cont += angulo
    angulos.append(cont)

totalperimetros = 0
veces = int(sys.argv[1])
distanciaPerimetros = []

#Generacion de Poligonos y Calculo de Perimetros
totalGeneracionPoligonos = 0;
totalCalculoPerimetros = 0;

for j in range(veces):
    pointsx = []
    pointsy = []
    #Generacion de Poligonos
    inicioGeneracionPoligonos = time.time()
    for angle in angulos:
        randomNumber_alterar = 0.01 + random.random() * 0.99
        x = int(randomNumber_alterar * radius * math.cos(math.radians(angle)))
        y = int(randomNumber_alterar * radius * math.sin(math.radians(angle)))
        pointsx.append(x)
        pointsy.append(y)
    finGeneracionPoligonos = time.time()
    totalGeneracionPoligonos += (finGeneracionPoligonos - inicioGeneracionPoligonos)

    #Calculo de Perimetros
    inicioCalculoPerimetros = time.time()
    sumPerimetros = 0
    for i in range(len(pointsx)):
        xUno, xDos, yUno, yDos, xCoordinate, yCoordinate, xSquared, ySquared, insideSum, squareRoot = 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
        xUno = pointsx[i]
        yUno = pointsy[i]
        if i == len(pointsx) - 1:
            xDos = pointsx[0]
            yDos = pointsy[0]
        else:
            xDos = pointsx[i + 1]
            yDos = pointsy[i + 1]

        #print("Coordenadas {}: ({}, {}) and ({}, {})".format(i, xUno, yUno, xDos, yDos))        
        xCoordinate = xDos - xUno
        yCoordinate = yDos - yUno
        xSquared = xCoordinate * xCoordinate
        ySquared = yCoordinate * yCoordinate
        insideSum = xSquared + ySquared
        squareRoot = math.sqrt(insideSum)
        distanciaPerimetros.append(squareRoot)
        #print("Distancia {}: {}".format(i, squareRoot))
    finCalculoPerimetros = time.time()
    totalCalculoPerimetros += (finCalculoPerimetros - inicioCalculoPerimetros)
    ##print("Fin del Poligono ", j+1, "\n")

sumPerimetros = sum(distanciaPerimetros)
avgPerimetros = sumPerimetros / len(distanciaPerimetros)

print("Suma Total: ", sumPerimetros)
print("El Promedio de los Perimetros Es:", avgPerimetros, "\n")

finEjecucion = time.time()
totalEjecucion = (finEjecucion - inicioEjecucion)

print("Tiempo de Generacion de Poligonos: " , (totalGeneracionPoligonos*1000) , " milisegundos")
print("Tiempo de Calculo de Todos los Perimetros: " , (totalCalculoPerimetros*1000) , " milisegundos")
print("Tiempo Total de Ejecucion: " , (totalEjecucion*1000) , " milisegundos")

File: Ejercicio 1/Poligonos/src/poligonos/Poligonos.java
package poligonos;

import java.util.Random;
import java.util.ArrayList;

public class Poligonos {

    public static void main(String[] args) {

        long inicioEjecucion = System.currentTimeMillis(); //Inicio de Ejecucion

        Random random = new Random(); //Variables para Generar los Angulos
        int radius = 200;
        double puntos = 10;
        double angulo = 360 / Integer.parseInt(args[1]);
        double cont = 0;
        ArrayList<Double> angulos = new ArrayList<>();

        angulos.add(cont);
        for (int i = 0; i < puntos; i++) {
            cont += angulo;
            angulos.add(cont);
        }

        ArrayList<Double> distanciaPerimetros = new ArrayList<>();

        //Generacion de Poligonos y Calculo de Perimetros
        long totalGeneracionPoligonos = 0;
        long totalCalculoPerimetros = 0;

        for (int i = 0; i < Integer.parseInt(args[0]); i++) {
            ArrayList<Integer> pointsx = new ArrayList<>();
            ArrayList<Integer> pointsy = new ArrayList<>();

             //Generacion de Poligonos
            long inicioGeneracionPoligonos = System.currentTimeMillis();
            for (double angle : angulos) {
                double randomNumber_alterar = 0.01 + random.nextDouble() * 0.99;
                int x = (int) (randomNumber_alterar * radius * Math.cos(Math.toRadians(angle)));
                int y = (int) (randomNumber_alterar * radius * Math.sin(Math.toRadians(angle)));
                x += 200;
                y += 200;
                pointsx.add(x);
                pointsy.add(y);
            }
            long finGeneracionPoligonos = System.currentTimeMillis();
            totalGeneracionPoligonos += (finGeneracionPoligonos - inicioGeneracionPoligonos);

            //Calculo de Perimetros
            long inicioCalculoPerimetros = System.currentTimeMillis();
            for (int j = 0; j < pointsx.size(); j++) {
                double xUno, xDos, yUno, yDos, xCoordinate, yCoordinate, xSquared, ySquared, insideSum, squareRoot;
                xUno = pointsx.get(j);
                yUno = pointsy.get(j);
                if (j == pointsx.size() - 1) {
                    xDos = pointsx.get(0);
                    yDos = pointsy.get(0);
                } else {
                    xDos = pointsx.get(j + 1);
                    yDos = pointsy.get(j + 1);
                }
                System.out.println("Coordenadas " + j + ": (" + (xUno - 200) + ", " + (yUno - 200) + "), (" + (xDos - 200) + ", " + (yDos - 200) + ")");
                xCoordinate = xDos - xUno;
                yCoordinate = yDos - yUno;
                xSquared = xCoordinate * xCoordinate;
                ySquared = yCoordinate * yCoordinate;
                insideSum = xSquared + ySquared;
                squareRoot = Math.sqrt(insideSum);

                distanciaPerimetros.add(squareRoot);
                System.out.println("Distancia " + j + ": " + squareRoot);
            }
            long finCalculoPerimetros = System.currentTimeMillis();
            totalCalculoPerimetros += (finCalculoPerimetros - inicioCalculoPerimetros);

            System.out.println("Fin del Poligono " + (i+1) + '\n');
        }

        double sumPerimetros = 0;
        for (int i = 0; i < distanciaPerimetros.size(); i++) {
            sumPerimetros += distanciaPerimetros.get(i);
        }
        double avgPerimetros = 0;
        avgPerimetros = sumPerimetros / distanciaPerimetros.size();
        System.out.println("Suma Total: " + sumPerimetros);
        System.out.println("El Promedio de los Perimetros Es: " + avgPerimetros + '\n');

        long finEjecucion = System.currentTimeMillis();
        long totalEjecucion = (finEjecucion - inicioEjecucion);

        System.out.println("Tiempo de Generacion de Poligonos: " + (totalGeneracionPoligonos) + " milisegundos");
        System.out.println("Tiempo de Calculo de Todos los Perimetros: " + (totalCalculoPerimetros) + " milisegundos");
        System.out.println("Tiempo Total de Ejecucion: " + totalEjecucion + " milisegundos");
    }
}